domain/model/mesh_results.py

```diff
@@ -105,21 +105,27 @@ def connection(self, from_agent, to_agent: AgentID) -> MeshColumn:
 
 
 class MeshResults:
-    """Internal representation of Mesh Test results; independent of source data structure like http or grpc synthetics client"""
+    """
+    Internal representation of Mesh Test results; independent of source data structure like http
+    or grpc synthetics client
+    """
 
     def __init__(
         self,
-        utc_timestamp: Optional[datetime] = None,
-        rows: List[MeshRow] = [],
+        utc_timestamp: datetime,
+        rows: Optional[List[MeshRow]] = None,
```

Comment:
Why Optional?

Reply:
So, it can be initialized to `None`. Initializing to an empty list (or any other mutable object) is undesirable, because it creates potential for multiple instances sharing the same instance (see [problem description](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument) and [canonical solution](https://docs.python.org/3/reference/compound_stmts.html#function-definitions)).